baseapp/baseapp.go

```diff
@@ -483,8 +483,8 @@ func handleQueryCustom(app *BaseApp, path []string, req abci.RequestQuery) (res
 // BeginBlock implements the ABCI application interface.
 func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeginBlock) {
 	if app.cms.TracingEnabled() {
-		app.cms.ResetTraceContext()
-		app.cms.WithTracingContext(sdk.TraceContext(
+		app.cms.SetTracingContext(nil)
```

Comment:
why we need to set the context to `nil` if we set it to `sdk.TraceContext` the next line?